README.md

```diff
@@ -201,6 +201,38 @@ Note that `#count` is calculated automatically based on the number of batches in
 your collection, and your Task’s progress will be displayed in terms of batches
 (not the number of records in the relation).
 
+On databases with large tables, sometime using an `ActiveRecord::Relation` to define a collection will
+cause your queries to timeout. One pattern to to have performant tasks is to look up the range of IDs that
+need to be operated on, then create a collection of ranges to query your models. For example:
+
+```ruby
+module Maintenance
+  class BackfillALotOfPosts < MaintenanceTasks::Task
+    # Generates an array like
+    # [(1..1_000_000),(1_000_001..2_000_000),(2_000_001..3_000_000), ...]
+    def collection
+      ranges = []
+      100.times do |i|
+        range_start = (1_000_000 * i) + 1
+        range_end = 1_000_000 * (i + 1)
+        ranges << (range_start..range_end)
+      end
+      ranges
```

Comment:
Wonder if we should consider recommending building "static" collections in a class-level memoized instance variable since we don't expect the collection to change between executions. It could also be a constant but there is no reason to slow down the boot time 

```suggestion
      100.times.map do |i|
        range_start = (1_000_000 * i) + 1
        range_end = 1_000_000 * (i + 1)
        (range_start..range_end)
      end
```

Reply:
Which also made me think that the collection can be as simple as:

`(0..100).to_a` and the `process` method could build the "current range" on-demand:

```ruby
def process(cursor_position)
        range_start = (1_000_000 * cursor_position) + 1
        range_end = 1_000_000 * (cursor_position + 1)
        range = range_start..range_end
end
```

Unlikely it will have any significant impact on performance but it should definitely help with making the collection as small as possible

Reply:
>Which also made me think that the collection can be as simple as:

I don't understand the internals well enough to understand why we want a collection that's as small as possible...especially since, practically speaking, an array of hundreds/thousands of ranges is likely not going to be a big deal compared to a collection of billions of records, but I defer to you on that as a neophyte 🙈

Reply:
Also, how embarrassing that I didn't use `map` 😂 

I've updated that bit of code to be less crappy 😂